tmt/steps/prepare/feature.py

```diff
@@ -79,6 +80,23 @@ def disable(self) -> None:
     }
 
 
+class FIPS(ToggleableFeature):
+    NAME = 'fips'
+
+    def is_unsupported(self) -> bool:
+        return isinstance(self.guest, (tmt.steps.provision.local.GuestLocal,
+                          tmt.steps.provision.podman.GuestContainer))
```

Comment:
Class enumeration will fail as soon as someone comes up with their own plugin that would not support this operation. I'd recommend adding something like `localhost` attribute to `Guest` class, see https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/local.py#L21 - it's set to `False` by default, `localhost` plugins sets it to `True`, and code checks this flag - not the class - when it's deciding about localhost-ish features. So I guess we need something like `fips_support` or something similar, set it to `True` by default & `False` in these two classes.

Reply:
Let me think about it. Adding a new flag might not be needed in the end since there is basically an equivalence between fips_support and "is_not_container" and I can most likely detect that in the playbook as well by inspecting init process control groups. That might be more tmt-friendly solution.

Reply:
Just to be sure, will such autodetection work with Image mode, a container-ish runtime?

We might add the `is_container` or similar flag to `Guest` classes if it turns out it's needed for more than just FIPS. I'd like all involved plugins to see the same picture, i.e. one plugin detecting a fact one way while another one would use completely different steps to discover the same answer would be something I'd wish to address. Not a blocker for now though, just a thought.

Reply:
So I actually add the support for enabling FIPS in containers too. The way how it "unofficially" works is that in a container you need the host kernel to be running in FIPS mode so the playbook only handles the userspace part. There are checks in the end of the playbook that the FIPS mode is correctly enabled and if the host is running in FIPS mode already it will pass and fail otherwise.

The way how it works is that the /proc/1/environ drives the detection of a container environment. By container I mean anything that does not have a control over kernelspace. I tried it in actual containers but I did not tried it anywhere else. I am not familiar with Image mode, is there some reasonably easy way to get a shell there to check /proc/1/environ content?